include/z64cutscene.h

```diff
@@ -3,286 +3,478 @@
+    /* 0xFFFF */ CS_CMD_END = 0xFFFF
+} CutsceneCmd;
 
-typedef struct {
-    /* 0x0 */ CutsceneCameraPoint* atPoints;
-    /* 0x4 */ CutsceneCameraPoint* eyePoints;
-    /* 0x8 */ s16 relativeToPlayer;
-} CutsceneCameraMove; // size = 0xC
+#define CS_CMD_CONTINUE 0
+#define CS_CMD_STOP -1
+
+typedef enum {
+    /* 0x00 */ CS_MISC_UNIMPLEMENTED_0,
+    /* 0x01 */ CS_MISC_RAIN,
+    /* 0x02 */ CS_MISC_LIGHTNING,
+    /* 0x03 */ CS_MISC_SET_CSFLAG_0, // also sets flag 2 if in Temple of Time
+    /* 0x04 */ CS_MISC_UNIMPLEMENTED_4,
+    /* 0x05 */ CS_MISC_UNIMPLEMENTED_5,
+    /* 0x06 */ CS_MISC_LIFT_FOG,
+    /* 0x07 */ CS_MISC_CLOUDY_SKY,
+    /* 0x08 */ CS_MISC_FADE_KOKIRI_GRASS_ENV_ALPHA,
+    /* 0x09 */ CS_MISC_SNOW,
+    /* 0x0A */ CS_MISC_SET_CSFLAG_1,
+    /* 0x0B */ CS_MISC_DEKU_TREE_DEATH,
+    /* 0x0C */ CS_MISC_STOP_CUTSCENE,
+    /* 0x0D */ CS_MISC_TRIFORCE_FLASH,
+    /* 0x0E */ CS_MISC_SET_LOCKED_VIEWPOINT,
+    /* 0x0F */ CS_MISC_SHOW_TITLE_CARD,
+    /* 0x10 */ CS_MISC_QUAKE_START,
+    /* 0x11 */ CS_MISC_QUAKE_STOP,
+    /* 0x12 */ CS_MISC_STOP_STORM_AND_ADVANCE_TO_DAY,
+    /* 0x13 */ CS_MISC_SET_FLAG_FAST_WINDMILL,
+    /* 0x14 */ CS_MISC_SET_FLAG_WELL_DRAINED,
+    /* 0x15 */ CS_MISC_SET_FLAG_LAKE_HYLIA_RESTORED,
+    /* 0x16 */ CS_MISC_VISMONO_BLACK_AND_WHITE,
+    /* 0x17 */ CS_MISC_VISMONO_SEPIA,
+    /* 0x18 */ CS_MISC_HIDE_ROOM,
+    /* 0x19 */ CS_MISC_TIME_ADVANCE_TO_NIGHT,
+    /* 0x1A */ CS_MISC_SET_TIME_BASED_LIGHT_SETTING,
+    /* 0x1B */ CS_MISC_RED_PULSATING_LIGHTS,
+    /* 0x1C */ CS_MISC_HALT_ALL_ACTORS,
+    /* 0x1D */ CS_MISC_RESUME_ALL_ACTORS,
+    /* 0x1E */ CS_MISC_SET_CSFLAG_3,
+    /* 0x1F */ CS_MISC_SET_CSFLAG_4,
+    /* 0x20 */ CS_MISC_SANDSTORM_FILL,
+    /* 0x21 */ CS_MISC_SUNSSONG_START,
+    /* 0x22 */ CS_MISC_FREEZE_TIME,
+    /* 0x23 */ CS_MISC_LONG_SCARECROW_SONG
+} CutsceneMiscType;
+
+typedef enum {
+    /* 0x00 */ CS_TEXT_NORMAL,
+    /* 0x01 */ CS_TEXT_CHOICE,
+    /* 0x02 */ CS_TEXT_OCARINA_ACTION,
+    /* 0x03 */ CS_TEXT_GORON_RUBY, // use `altTextId1` in the ruby cutscene if sapphire is already obtained
+    /* 0x04 */ CS_TEXT_ZORA_SAPPHIRE // use `altTextId1` in the sapphire cutscene if ruby is already obtained
+} CutsceneTextType;
+
+typedef enum {
+    /* 0x03 */ CS_FADE_OUT_FANFARE = 3, 
+    /* 0x04 */ CS_FADE_OUT_BGM_MAIN
+} CutsceneFadeOutSeqPlayer;
+
+typedef enum {
+    /* 0x01 */ CS_TRANS_GRAY_FILL = 1, // has hardcoded sounds for some scenes
+    /* 0x02 */ CS_TRANS_BLUE_FILL,
+    /* 0x03 */ CS_TRANS_RED_UNFILL,
+    /* 0x04 */ CS_TRANS_GREEN_UNFILL,
+    /* 0x05 */ CS_TRANS_GRAY_UNFILL,
+    /* 0x06 */ CS_TRANS_BLUE_UNFILL,
+    /* 0x07 */ CS_TRANS_RED_FILL,
+    /* 0x08 */ CS_TRANS_GREEN_FILL,
+    /* 0x09 */ CS_TRANS_TRIGGER_INSTANCE, // used with `TRANS_MODE_INSTANCE_WAIT`
+    /* 0x0A */ CS_TRANS_BLACK_UNFILL,
+    /* 0x0B */ CS_TRANS_BLACK_FILL,
+    /* 0x0C */ CS_TRANS_BLACK_HALF_FILL, // used with `TRANS_MODE_CS_BLACK_FILL`
+    /* 0x0D */ CS_TRANS_BLACK_FILL_FROM_HALF
```

Comment:
```suggestion
    /* 0x0C */ CS_TRANS_BLACK_UNFILL_TO_HALF, // used with `TRANS_MODE_CS_BLACK_FILL`
    /* 0x0D */ CS_TRANS_BLACK_FILL_FROM_HALF
```
:thinking: 
this means all the other `TRANS_MODE_CS_BLACK_FILL` related names are off?

(I guess this is used in the intro cutscene when link is sleeping and having the nightmare)

Reply:
yeah it is the nightmare in links house. the half fill makes the room look dark.
Im pretty lost with the naming of these things now though, what needs to change?

Reply:
As established:

With `CS_TRANS_BLACK_HALF_FILL`(->`CS_TRANS_BLACK_UNFILL_TO_HALF`) the screen goes from 255-alpha (opaque) black to 100-alpha black

With `CS_TRANS_BLACK_FILL_FROM_HALF` the screen goes from 100-alpha black to 255-alpha (opaque) black

And since `CS_TRANS_BLACK_HALF_FILL`(->`CS_TRANS_BLACK_UNFILL_TO_HALF`) interacts through `gSaveContext.cutsceneTransitionControl` with `TRANS_MODE_CS_BLACK_FILL` (I assume),
that challenges the names of `TRANS_MODE_CS_BLACK_FILL` and the related `TRANS_MODE_CS_BLACK_FILL_INIT` and `TRANS_TYPE_CS_BLACK_FILL`

<img src="https://cdn.discordapp.com/emojis/681587643993161762.webp?size=128&quality=lossless" alt="Pepega" height="32">

Reply:
the z_play/transition side only really knows that the screen is being filled with a black color, so `TRANS_MODE_CS_BLACK_FILL` and co seem fine to me. Its the manipulation of `cutsceneTransitionControl` that z_demo does that matters

z_play currently uses "fill in" and "fill out" for fill/unfill, so maybe using that terminology in z_demo will make things easier to understand (in that, "fill" on its own just means the screen is filled somehow, not indicating any direction)

Reply:
made some changes, see how that looks

Reply:
basically `TRANS_MODE_CS_BLACK_FILL` from z_play just means "cutscene variable controls the fill"
and in/out describe the direction z_demo side `CS_TRANS_BLACK_FILL_OUT_TO_HALF` / `CS_TRANS_BLACK_FILL_IN_FROM_HALF` 

and this change in general is consistent with how z_play describes its fills

Reply:
looks good :+1: